dipy/tracking/propspeed.pyx

```diff
@@ -89,73 +88,104 @@ def ndarray_offset(cnp.ndarray[cnp.npy_intp, ndim=1] indices, \
     >>> A.ravel()[4]==A[1,1]
     True
     '''
-    return offset(<cnp.npy_intp*>indices.data,<cnp.npy_intp*>strides.data,lenind, typesize)
+    if not cnp.PyArray_CHKFLAGS(indices, cnp.NPY_C_CONTIGUOUS):
+        raise ValueError(u"indices is not C contiguous")
+    if not cnp.PyArray_CHKFLAGS(strides, cnp.NPY_C_CONTIGUOUS):
+        raise ValueError(u"strides is not C contiguous")
+    return offset(<cnp.npy_intp*> cnp.PyArray_DATA(indices),
+                  <cnp.npy_intp*> cnp.PyArray_DATA(strides),
+                  lenind,
+                  typesize)
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def map_coordinates_trilinear_iso(cnp.ndarray[double, ndim=3] data,\
-                                   cnp.ndarray[double, ndim=2] points,\
-                                   cnp.ndarray[cnp.npy_intp, ndim=1] data_strides,\
-                                   cnp.npy_intp len_points,\
-                                   cnp.ndarray[double, ndim=1] result):
-    ''' trilinear interpolation (isotropic voxel size)
+def map_coordinates_trilinear_iso(
+    cnp.ndarray[double, ndim=3] data,
```

Comment:
Yeah, I see that data_strides goes a bit out. But look below too. There are other places that this is not the case.